tests/core/test_expectation_suite.py

```diff
@@ -312,6 +311,145 @@ def test_add_citation_with_profiler_config(
+                return False
+
+        suite1 = ExpectationSuite("suite_1", expectations=[StubExpectationConfiguration()])  # type: ignore[arg-type]
+        suite2 = ExpectationSuite("suite_2", expectations=[StubExpectationConfiguration()])  # type: ignore[arg-type]
+        assert not suite1.isEquivalentTo(suite2)
+
+    @pytest.mark.unit
+    def test_is_equivalent_to_expectation_suite_and_dict_true(
+        self, suite_with_single_expectation: ExpectationSuite
+    ):
+        """Suite should be equivalent to its dict representation."""
+        suite_with_single_expectation_dict: dict = expectationSuiteSchema.dump(
+            suite_with_single_expectation
+        )
+
+        assert suite_with_single_expectation.isEquivalentTo(
+            suite_with_single_expectation_dict
+        )
+
+    @pytest.mark.unit
+    def test_is_equivalent_to_expectation_suite_and_dict_false(
+        self, suite_with_single_expectation: ExpectationSuite
+    ):
+        modified_suite = deepcopy(suite_with_single_expectation)
+        modified_suite.expectations[0]["kwargs"]["value_set"][0] = -1
+
+        modified_suite_dict: dict = expectationSuiteSchema.dump(modified_suite)
+
+        assert not suite_with_single_expectation.isEquivalentTo(modified_suite_dict)
+
+    @pytest.mark.unit
+    def test_is_equivalent_to_invalid_expectation_config_dict_returns_notimplemented(
+        self, suite_with_single_expectation: ExpectationSuite
+    ):
+        return_value = suite_with_single_expectation.isEquivalentTo(
+            {"not_valid": "expectation_suite_config_dict"}
+        )
+        assert return_value == NotImplemented
+
+    @pytest.mark.unit
+    def test_is_equivalent_to_unsupported_class_returns_notimplemented(
+        self, suite_with_single_expectation: ExpectationSuite
+    ):
+        """If we are not comparing to an ExpectationConfiguration or dict then we should return NotImplemented."""
+
+        class UnsupportedClass:
+            pass
+
+        return_value = suite_with_single_expectation.isEquivalentTo(UnsupportedClass())
+        assert return_value == NotImplemented
+
+    @pytest.mark.integration
+    def test_is_equivalent_to_expectation_suite_classes_true_with_changes_to_non_considered_attributes(
+        self, suite_with_single_expectation: ExpectationSuite
+    ):
+        """Only expectation equivalence is considered for suite equivalence. Marked as integration since this uses the ExpectationConfiguration.isEquivalentTo() under the hood."""
+        different_but_equivalent_suite = deepcopy(suite_with_single_expectation)
+        different_but_equivalent_suite.expectation_suite_name = "different_name"
+        different_but_equivalent_suite.meta = {"notes": "Different meta."}
+        different_but_equivalent_suite.data_asset_type = "different_data_asset_type"
+        different_but_equivalent_suite.ge_cloud_id = "different_id"
+
+        assert suite_with_single_expectation.isEquivalentTo(
```

Comment:
Ok cool - I thought it would be an integration test on the ExpectationSuite since it is depending also on the ExpectationConfiguration. Where the related test (`test_is_equivalent_to_expectation_suite_classes_true_with_changes_to_non_considered_attributes_unit_test`) is a unit test since it stubs `ExpectationConfiguration.isEquivalentTo()` in the test for `ExpectationSuite.isEquivalentTo()`